components/AvalancheForecastZoneMap.tsx

```diff
@@ -53,154 +48,94 @@ export const AvalancheForecastZoneMap: React.FunctionComponent<MapProps> = ({cen
-  if (isLoading) {
-    return (
-      <FlatList
-        horizontal={true}
-        data={[{id: center_id, title: center_id}]}
-        renderItem={({item}) => (
-          <View style={cardStyle}>
-            <Text>{item.title}</Text>
-            <ActivityIndicator />
-          </View>
-        )}
-      />
-    );
-  }
-  if (isError) {
-    return (
-      <FlatList
-        horizontal={true}
-        data={[{id: center_id, title: center_id}]}
-        renderItem={({item}) => (
-          <View style={cardStyle}>
-            <Text>{item.title}</Text>
-            <Text>{`Could not fetch forecast zones for ${center_id}: ${error?.message}.`}</Text>
-            <ActivityIndicator />
-          </View>
-        )}
-      />
-    );
-  }
 
-  let props = {};
-  if (horizontal) {
-    props = {
-      horizontal: true,
-      snapToAlignment: 'start',
-      decelerationRate: 'fast',
-      snapToOffsets: mapLayer?.features.filter(feature => feature.type === 'Feature').map((feature, index) => index * CARD_WIDTH * width + (index - 1) * CARD_SPACING * width),
-      contentInset: {
-        left: CARD_MARGIN * width,
-        right: CARD_MARGIN * width,
-      },
-      contentContainerStyle: {paddingHorizontal: CARD_MARGIN * width},
-    };
-  }
+  const props = {
+    snapToAlignment: 'start',
+    decelerationRate: 'fast',
+    snapToOffsets: mapLayer?.features.filter(feature => feature.type === 'Feature').map((feature, index) => index * CARD_WIDTH * width + (index - 1) * CARD_SPACING * width),
+    contentInset: {
+      left: CARD_MARGIN * width,
+      right: CARD_MARGIN * width,
+    },
+    contentContainerStyle: {paddingHorizontal: CARD_MARGIN * width},
+  } as const;
 
   return (
     <FlatList
+      horizontal
+      width="100%"
+      position="absolute"
+      bottom="4"
       {...props}
       data={mapLayer?.features
         .filter(feature => feature.type === 'Feature')
         .map(feature => ({
-          id: center_id + feature.id,
+          id: center + feature.id,
           feature: feature,
           date: date,
         }))}
-      renderItem={({item}) => <AvalancheForecastZoneCard key={item.id} feature={item.feature} date={item.date} style={cardStyle} />}></FlatList>
+      renderItem={({item}) => <AvalancheForecastZoneCard key={item.id} feature={item.feature} date={item.date} />}></FlatList>
   );
 };
 
-export const AvalancheForecastZoneCard: React.FunctionComponent<{
+const AvalancheForecastZoneCard: React.FunctionComponent<{
   feature: Feature;
   date: string;
-  style: ViewStyle;
-}> = ({feature, date, style}) => {
-  const forecastDate: Date = parseISO(date);
+}> = ({feature, date}) => {
   const {width} = useWindowDimensions();
   const navigation = useNavigation<HomeStackNavigationProps>();
-  const {isLoading, isError, data: forecast, error} = useAvalancheForecastFragment(feature.properties.center_id, feature.id, forecastDate);
```

Comment:
Also - the reason I had done it this way is:

- other than this transient data, everything else in the center metadata is effectively static, meaning we can safely cache it for very long periods of time
- if we derive this part of the map from the data in the map layer, in order to keep the entire app in sync we will need to methodically refresh the map layer every time we refresh the avalanche forecast, whether that's a background refresh or driven by push notifications

It seems more efficient and likely easier to ensure we don't make mistakes if we paint this based on the avalanche forecast API itself, but the dichotomy between `v2/public/products` (`useAvalancheForecastFragment`) and `v2/public/product` (`useAvalancheForecast`) has similar source-of-truth issues. 

We should also think about what happens if a user's in a forecast view, refreshes to get an updated forecast, and goes back to the map - ideally, they get the updated forecast colored in to the zone. 

Probably good to file this one away for thinking about more later, the APIs here do not give us what we need, really.

Reply:
But this was pulling N requests just to get the transient data for each zone, rather than 1 request for the whole center. It feels like a no-brainer to remove these requests from this view. What am I missing? Even if we pre-cache the map regions and just pull the danger level from the response, it's still way less resource intensive.

Pre-loading the detailed forecasts optimistically seems like a worthy thing, but not at the expense of slowing this component down.

Reply:
And even if we decide to go back to N requests, I think it's probably saner to group the N requests with `Promise.all` and treat the whole group as succeeding or failing. Having individual cards in success/loading/error states feels like the wrong level of specificity for end-users.

Reply:
> We should also think about what happens if a user's in a forecast view, refreshes to get an updated forecast, and goes back to the map - ideally, they get the updated forecast colored in to the zone.

Agreed! Things like this have got me thinking about integrating Redux or some other state management library. Redux would also be a clean way to boot up the app with data that we know doesn't change much, like map region data, and still allow for filling in changes dynamically based on network requests.

Reply:
I might be wrong here, but the beauty of `react-query` is that you get the state management for free and there's no need whatsoever for more management on top. We can set the caching behavior for the map layer to have a TTL of a month or something similar.

The issue with pulling forecast data from the map layer or the many-products API is just a source-of-truth thing, as I noted - we will need to be very careful to keep our caches coherent.

Reply:
Request de-duplication bug fixed in https://github.com/stevekuznetsov/avalanche-forecast/pull/52

Reply:
I was thinking more about your comments about slowing the component down and leading to a bad UX when only one zone was failing to load, and I think there might also be a fairly straightforward way to achieve that - the `useAvalancheForecastFragment()` hook is simply filtering the data from `useAvalancheForecastFragments()` - which applies to all zones in the center. If we pulled that state up one level, we'd have to handle the entire set of fragments being missing or not, but never any individual one. In practice, we'd of course never see an individual one missing since there's only one network request underneath it all, but you are 100% right that all the extra error handling code for each zone is completely unnecessary. 

WDYT? Would such an approach provide the better UX you are looking for?

Reply:
Is your goal ultimately to never call `/v2/public/products/map-layer`? I guess I'm not understanding what's driving you here, maybe we can dig into that in our chat.

> Notably, the two layers of requests remain required to display the forecast details- take a look at the useAvalancheForecast flow - there is no way to get the ID of the latest forecast for a zone without asking for all the forecasts for that zone during some period of time as the map layer leaves out this crucial bit of information.

I could also use an explainer here because it seems like you can always just fetch the latest forecast for a given `center_id` and `zone_id`. I think I'm missing something!

```
❯ curl 'https://api.avalanche.org/v2/public/product?type=forecast&center_id=NWAC&zone_id=1131' | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9835    0  9835    0     0  65702      0 --:--:-- --:--:-- --:--:-- 71788
{
  "id": 116746,
  "published_time": "2023-01-09T02:00:00+00:00",
  "expires_time": "2023-01-10T02:00:00+00:00",
  "created_at": "2023-01-09T01:36:29+00:00",
  "updated_at": "2023-01-09T01:40:45+00:00",
  "author": "Dallas Glass",
  "product_type": "forecast",
  // etc...
```